source/guide_buildbot.rst

```diff
@@ -0,0 +1,266 @@
+
+If pip doesn't report the new version right away, try logging out and back in again.
+
+Installation
+============
+
+You can install the BuildBot_ bundle using pip:
+
+::
+
+ [isabell@stardust ~]$ pip install buildbot[bundle] --user
+
+If you want e-mail notifications from BuildBot_ to work later on, you will also need to install ``pyopenssl`` and ``service-identity``:
+
+::
+
+ [isabell@stardust ~]$ pip install pyopenssl service-identity --user
+
+
+Configuration of the master
+===========================
+
+Step 1
+------
+Prepare the BuildBot_ folders. We are going to use one folder for all of our masters and another for all the workers:
+
+::
+
+ [isabell@stardust ~]$ mkdir ~/bb-master ~/bb-workers
+
+Step 2
+------
+
+Create the first master:
+
+::
+
+ [isabell@stardust ~]$ cd ~/bb-master
+ [isabell@stardust bb-master]$ buildbot create-master master
+
+This will create the directory ``bb-master/master``, set up a sqlite database and deposit a sample configuration.
+
+Step 3
+------
+
+Move the sample configuration:
+
+::
+
+ [isabell@stardust bb-master]$ mv master/master.cfg.sample master/master.cfg
+
+
+Step 4
+------
+
+For BuildBot_ to work, we need to find two open ports on your uberspace host. One for the master's webinterface and one for the workers to communicate with the master:
+
+::
+
+ [isabell@stardust bb-master] FREEPORT=$(( $RANDOM % 4535 + 61000 )); ss -ln src :$FREEPORT | grep $FREEPORT && echo "try again" || echo $FREEPORT
+ 12345
+ [isabell@stardust bb-master] FREEPORT=$(( $RANDOM % 4535 + 61000 )); ss -ln src :$FREEPORT | grep $FREEPORT && echo "try again" || echo $FREEPORT
+ 54321
+
+We'll use ``12345`` for the webinterface and ``54321`` for the worker connections. You will get other ports as results (something between 61000 and 65535).
+
+
+Step 5
+------
+
+Edit the file ``/home/isabell/bb-master/master/master.cfg``, which is basically a Python_ file. For now, we only need to change the ports. In ``c['www']``, change the port of the webinterface to ``12345`` (as selected before) and in ``c['protocols']``, change the port to ``54321``. That is going to be the port that the workers will communicate through. You should read through the rest of the options already, but leave things to their default values for now.
+
+Step 6
+------
+
+That's it! Our master should be able to start now:
+
+::
+
+ [isabell@stardust bb-master] buildbot start master
+ Following twistd.log until startup finished..
+ The buildmaster appears to have (re)started correctly.
+
+If you don't get the same output, check the log at ``master/twistd.log`` for errors.
+
+Configuration of the worker
+===========================
+
+Now that the master is done, let's create the first worker!
+
+Step 1
+------
+
+Change directories and create the worker:
+
+::
+
+ [isabell@stardust bb-master] cd ~/bb-workers
+ [isabell@stardust bb-workers] buildbot-worker create-worker example-worker localhost:54321 example-worker pass
```

Comment:
I was wondering the same thing - whether I should go into detail on how to configure the 'pipeline'. I decided against it because the "BuildBot in 5 minutes" guide over at their homepage does quite a nice job on explaining the mechanics and terminology.

Or do I misunderstand and you just want a notice for the user to point out and get rid of the example configuration?

Reply:
A [`.. note::`](http://docutils.sourceforge.net/docs/ref/rst/directives.html#specific-admonitions) explaining that there is default config and that further configuration advice can be found in the "BuildBot in 5 minutes" guide (linked) would be nice :)